modular_splurt/code/game/objects/items/lewd_items/chastity_hypno/chastity_hypnosis.dm

```diff
@@ -0,0 +1,145 @@
+/obj/item/chastity_hypno
+	name = "Chastity Hypno"
+	var/list/available_modes = list("None", "Impotent", "Edging-Only", "Disappointing-Orgasm", "Overstimulated", "Hyper-Sensitive")
+	var/list/choices = list("Anus" = "None", "Breasts" = "None", "Penis" = "None", "Vagina" = "None")
+
+/obj/item/chastity_hypno/proc/hypno(mob/target)
+	if(!iscarbon(target) || !choices)
+		return
+
+	var/mob/living/carbon/C = target
+
+	C.remove_chastity_hypno_effects()
+
+	for(var/G in choices)
+		if(choices[G] == "None")
+			continue
+
+		if(G == "Anus" && C.has_anus(REQUIRE_ANY))
+			var/trait_flag
+			switch(choices[G])
+				if("Impotent")
+					trait_flag = TRAIT_IMPOTENT_ANUS
+				if("Edging-Only")
+					trait_flag = TRAIT_EDGINGONLY_ANUS
+				if("Disappointing-Orgasm")
+					trait_flag = TRAIT_DISAPPOINTING_ANUS
+				if("Overstimulated")
+					trait_flag = TRAIT_OVERSTIM_ANUS
+				if("Hyper-Sensitive")
+					trait_flag = TRAIT_HYPERSENS_ANUS
+
+			ADD_TRAIT(C, trait_flag, ORGAN_TRAIT)
+			to_chat(C, "<span class='hypnophrase'>Your anus is now <i>[lowertext(choices[G])]</i></span>")
+			continue
+
+		var/obj/item/organ/genital/genital = C.getorganslot(lowertext(G))
+		if(!genital)
+			continue
+
+
+		var/hypno_flag
+
+		switch(choices[G])
+			if("Impotent")
+				hypno_flag = GENITAL_IMPOTENT
+				if(istype(genital, /obj/item/organ/genital/penis))
+					genital.set_aroused_state(0, "impotence") //Pp goes wooon
+			if("Edging-Only")
+				hypno_flag = GENITAL_EDGINGONLY
+			if("Disappointing-Orgasm")
+				hypno_flag = GENITAL_DISAPPOINTING
+			if("Overstimulated")
+				hypno_flag = GENITAL_OVERSTIM
+			if("Hyper-Sensitive")
+				hypno_flag = GENITAL_HYPERSENS
+
+		if(!hypno_flag)
+			continue
+
+		to_chat(C, "<span class='hypnophrase'>Your [lowertext(G)] is now <i>[lowertext(choices[G])]</i></span>")
+
+		var/obj/item/organ/genital/genital_organ = genital
+		ENABLE_BITFIELD(genital_organ.genital_flags, hypno_flag)
+
+	SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "hypno", /datum/mood_event/hypnosis)
+	choices = list("Anus" = "None", "Breasts" = "None", "Penis" = "None", "Vagina" = "None")
```

Comment:
In general, for all these hypno items that apply effects you'll want to first make sure that they check for the hypnosis pref, and also you'll wanna make preferences for if the player wants to actually be affected by them first before they can be used. The impotent and disappointing flags could check for the chastity pref, the overstim and hypersens ones could use their own pref for both and edginglonly would have its own pref as well
These new prefs should be bitflags in the cit_toggles variable.